### PR TITLE
Update CONTRIBUTING.md

### LINKED ISSUE #5487: CONTRIBUTING.md is outdated
- There is a desktop embedder section which is irrelevant
- pub get fails on flutter stable, the guide should mention that the local development of devtools should happen on a flutter master channel.

### PR DESCRIPTION
**Pull request by lesnitsky**
- drop desktop embedding note (slightly outdated 😉)
- add instruction to use flutter master channel (pub get fails due to incompatible `vm_service` version

## Related issues

Closes #5487

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or there is a reason for not adding tests.

![build.yaml badge]

If you need help, consider asking for help on [Discord].

[Contributor Guide]: https://github.com/flutter/devtools/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[build.yaml badge]: https://github.com/flutter/devtools/actions/workflows/build.yaml/badge.svg

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ You can do this online, and it only takes a minute.
 
 ## Workflow for making changes
 
+- Switch to flutter master channel `flutter channel master`
```
Comment (CoderDake) [inline CONTRIBUTING.md:21]:
```suggestion
- Change flutter to the latest flutter candidate:
    `./tool/update_flutter_sdk.sh --local`
```

Comment (CoderDake) [reply CONTRIBUTING.md:21]: This helper was made to switch to the version specified by ./tool/latest_flutter_candidate.sh, which is the version that we build Devtools against :D

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -126,24 +128,6 @@ This instructs VS Code to run the `tool/build_e2e.dart` script instead of runnin
 
 Next, restart VS Code (or run the **Developer: Reload Window** command from the command palette (`F1`)) and DevTools will be run from your local code. After making any code changes to DevTools or the server, you will need to re-run the **Developer: Reload Window** command to rebuild and restart the server.
 
-### Desktop Embedder
```
Comment (kenzieschmoll) [inline CONTRIBUTING.md:129]: Let's leave this section. Developing DevTools with Flutter desktop is a much nicer development experience.  Here's my suggested update.
```
### Desktop Embedder

You can also run DevTools using the Flutter desktop embedder on linux or macos. Some DevTools features only work on the web, like the embedded Perfetto trace viewer or DevTools analytics, but the limitations on the desktop app are few.

The advantage of developing with the desktop embedder is that you can use hot reload to speed up your development cycle.

To run DevTools with the desktop embedder, you can run with either of the following from `devtools/packages/devtools_app`:

- `flutter run -d macos`
- `flutter run -d linux`

If this  fails, you may need to run `flutter create .` from `devtools/packages/devtools_app` to generate the updated files for your platform.
```

**Comment by Hixie** [comment]
@lesnitsky is this still on your radar?

**Review by kenzieschmoll — APPROVED** [review]

**Comment by kenzieschmoll** [comment]
I think you'll need to merge with master to resolve the issues on the bots.

**Comment by lesnitsky** [comment]
I've rebased master

**Pull request merged**